x-pack/platform/plugins/shared/search_connectors/public/api/index/create_api_index_api_logic.ts

```diff
@@ -0,0 +1,54 @@
+/*
```

Comment:
Are you sure we need this file? I don't think this logic should me migrated.

Reply:
Yes, because it is used by Native connector configuration by AttachIndexBox https://github.com/elastic/kibana/blob/b9264e57f92a2b9ce1e25ce3f791282b36a92337/x-pack/platform/plugins/shared/search_connectors/public/components/connector_detail/attach_index_box.tsx#L61-L63

Reply:
🤔  I'll take a look, on the surface that doesn't make sense. If we're attaching an index I _think_ it's supposed to exist already. And if you are creating the index when you add the connector it does the creation on the backend. I don't _think_ the createIndex api logic is every actually used even if we're still referencing it. It may just have not been cleaned up when we stopped using it.

Reply:
So where this is used in [attach_index_box](https://github.com/elastic/kibana/blob/b9264e57f92a2b9ce1e25ce3f791282b36a92337/x-pack/platform/plugins/shared/search_connectors/public/components/connector_detail/attach_index_box.tsx#L353) Is only if the connector doesn't have an index name. So it shouldn't currently ever be rendered.

If it was this would break as that `createIndex` action and this `create_api_index_api_logic` are referencing endpoints that do not exist in `search_connectors`.

This is because this new connectors flow creates the index in the backend so a connector should never not have an index. Which makes the whole `attach_index_box` component a but useless IMO, but thats not directly related to this work.

Reply:
cleaned up by the [commit](https://github.com/elastic/kibana/pull/213509/commits/b03b5e02ae2b5f83413a34c9a22fa4e055c65a9c)